apps/aeutil/lib/serialization.ex

```diff
@@ -10,34 +10,144 @@ defmodule Aeutil.Serialization do
-      chain_state_hash: bech32_binary(block.header.chain_state_hash, :chainstate, direction),
-      prev_hash: bech32_binary(block.header.prev_hash, :header, direction),
-      txs_hash: bech32_binary(block.header.txs_hash, :txs, direction)}
-    new_txs = Enum.map(block.txs, fn(tx) -> tx(tx, direction) end)
+    new_header = %{
+      block.header
+      | chain_state_hash: bech32_binary(block.header.chain_state_hash, :chainstate, direction),
+        prev_hash: bech32_binary(block.header.prev_hash, :header, direction),
+        txs_hash: bech32_binary(block.header.txs_hash, :txs, direction)
+    }
+
+    new_txs =
+      Enum.map(block.txs, fn tx ->
+        map_to_tx(tx, direction)
+      end)
+
     Block.new(%{block | header: Header.new(new_header), txs: new_txs})
   end
 
+  @spec tx(SignedTx.t(), atom(), :serialize | :deserialize) :: SignedTx.t()
+  def tx(tx, type, direction) do
+    case type do
+      :spend_tx ->
+        new_data = %{
+          tx.data
+          | from_acc: hex_binary(tx.data.from_acc, direction),
+            to_acc: hex_binary(tx.data.to_acc, direction)
+        }
+
+        new_signature = hex_binary(tx.signature, direction)
+        %SignedTx{data: SpendTx.new(new_data), signature: new_signature}
+
+      :voting_tx ->
+        case tx.data do
+          %Aecore.Structures.VotingTx{data: %Aecore.Structures.VotingQuestionTx{}} ->
+            new_data = %{tx.data.data | from_acc: hex_binary(tx.data.data.from_acc, direction)}
+            new_signature = hex_binary(tx.signature, direction)
+
+            %SignedTx{
+              data: %VotingTx{data: VotingQuestionTx.new(new_data)},
+              signature: new_signature
+            }
+
+          %Aecore.Structures.VotingTx{data: %Aecore.Structures.VotingAnswerTx{}} ->
+            new_data = %{
+              tx.data.data
+              | from_acc: hex_binary(tx.data.data.from_acc, direction),
+                hash_question: hex_binary(tx.data.data.hash_question, direction)
+            }
+
+            new_signature = hex_binary(tx.signature, direction)
+
+            %SignedTx{
+              data: %VotingTx{data: VotingAnswerTx.new(new_data)},
+              signature: new_signature
+            }
+        end
+
+      _ ->
+        Logger.error("Unidentified type")
+    end
+  end
+
   @spec tx(SignedTx.t(), :serialize | :deserialize) :: SignedTx.t()
   def tx(tx, direction) do
-    new_data = %{tx.data |
-                 from_acc: hex_binary(tx.data.from_acc, direction),
-                 to_acc: hex_binary(tx.data.to_acc, direction)}
-    new_signature = base64_binary(tx.signature, direction)
+    new_data = %{
+      tx.data
+      | from_acc: hex_binary(tx.data.from_acc, direction),
+        to_acc: hex_binary(tx.data.to_acc, direction)
+    }
+
+    new_signature = hex_binary(tx.signature, direction)
     %SignedTx{data: SpendTx.new(new_data), signature: new_signature}
   end
 
+  @spec map_to_tx(map(), :serialize | :deserialize) :: SignedTx.t()
```

Comment:
`map_to_tx` indicates only `:deserialize` direction, shouldn't this functionality be included in the `tx` function